sql/catalyst/src/main/scala/org/apache/spark/sql/catalyst/util/DateTimeUtils.scala

```diff
@@ -167,6 +167,21 @@ object DateTimeUtils {
     ts
   }
 
+  /**
+   * Converts microseconds since the epoch to an instance of `java.sql.Timestamp`.
+   *
+   * @param micros The number of microseconds since 1970-01-01T00:00:00.000000Z.
+   * @return A `java.sql.Timestamp` from number of micros since epoch.
+   */
+  def toJavaTimestampNoRebase(micros: Long): Timestamp = {
```

Comment:
is it possible to share code with `toJavaTimestamp`?  e.g.
```
def toJavaTimestamp ... {
  toJavaTimestampNoRebase(rebaseGregorianToJulianMicros(micros))
}
```

Reply:
Got it.